source/routing/raptor_solution_reader.cpp

```diff
@@ -115,6 +121,60 @@ struct Journey {
             }
         }
     }
+    std::pair<const type::StopTime*, DateTime>
+    get_out_st_dt(const std::pair<const type::StopTime*, DateTime>& in_st_dt,
+                  const type::JourneyPatternPoint*const target_jpp) const
+    {
+        DateTime cur_dt = in_st_dt.second;
+        for (const auto& st: raptor_visitor().st_range(*in_st_dt.first).advance_begin(1)) {
+            const auto current_time = st.section_end_time(true, DateTimeUtils::hour(cur_dt));
+            DateTimeUtils::update(cur_dt, current_time, true);
+            if (st.journey_pattern_point == target_jpp) {
+                // check if the get_out is valid?
+                return {&st, cur_dt};
+            }
+        }
+        for (const auto* stay_in_vj = in_st_dt.first->vehicle_journey->next_vj;
+             stay_in_vj != nullptr;
+             stay_in_vj = stay_in_vj->next_vj) {
+            for (const auto& st: stay_in_vj->stop_time_list) {
+                const auto current_time = st.section_end_time(true, DateTimeUtils::hour(cur_dt));
+                DateTimeUtils::update(cur_dt, current_time, true);
+                if (st.journey_pattern_point == target_jpp) {
+                    // check if the get_out is valid?
+                    return {&st, cur_dt};
+                }
+            }
+        }
+        assert(false);
+        return {nullptr, 0};
+    }
+    template<typename Visitor>
+    void align_left(const RaptorSolutionReader<Visitor>& reader) {
+        if (sections.size() < 3) { return; }
+        const auto* prev_s = &sections.at(0);
+        for (auto& cur_s: boost::make_iterator_range(sections.begin() + 1, sections.end() - 1)) {
+            const auto* cur_jpp = cur_s.get_in_st->journey_pattern_point;
+            const auto* conn = reader.raptor.data.pt_data->get_stop_point_connection(
+                *prev_s->get_out_st->journey_pattern_point->stop_point,
+                *cur_jpp->stop_point);
```

Comment:
Do we have something to check this isn't possible to have 2 connections linking the same two stop_points?